_posts/en/newsletters/2019-11-13-newsletter.md

```diff
@@ -0,0 +1,367 @@
+    proof-of-work to try to prevent nodes from using using the extra
+    up-front payment information to guess the length of the route.
+    Anthony Towns proposed a partial [alternative][towns up-front]
+    focusing on managing payment amounts using a refund mechanism.
+
+    Joost Jager suggested that up-front payments should only be required
+    as a last resort because even small additional fees could make
+    micropayments uneconomic.  He suggested that it should be possible
+    to address bandwidth-wasting network activity using rate limits
+    based on node reputation and, further, that research into up-front
+    payments should focus on first solving liquidity abuse---where an
+    attacker ties up someone's in-channel funds for period of time---as
+    the solution to that problem may also prevent the abuse of routing
+    node bandwidth.
+
+    Ultimately, no conclusions were reached and discussion about the
+    topic remains ongoing as of this writing.
+
+- **Proposed BOLT for LN offers:** Rusty Russell [posted][bolt offers]
+  draft text for a new BOLT that would allow users to submit offers and
+  receive invoices over the LN routing protocol.  For example, Alice
+  could subscribe to a service provided by Bob where each month she
+  would submit an offer to pay Bob, Bob would reply with an invoice,
+  Alice would pay the invoice, and Bob would provide the service.
+
+    Early feedback on the proposal suggested that it might want to
+    incorporate by reference an established language for
+    machine-readable invoices, such as the [Universal Business
+    Language][] (UBL).  However, there was a concern that implementing
+    the full UBL specification would be an excessive burden on developers
+    of LN software.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[libsecp256k1][libsecp256k1 repo], [Bitcoin Improvement Proposals
+(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #16110][] adds support for building Bitcoin Core (including
+  the GUI) for the Android Native Development Kit (NDK).  In contrast to
+  independent projects like [Android Bitcoin Core][] (ABcore) that build
+  their own Bitcoin Core binaries for Android NDK, adding support
+  directly to the Bitcoin Core project may simplify testing and could
+  allow future PRs to add the Android builds to the set of
+  deterministically-generated executables that allow users to receive
+  greater assurances that they are running the same well-reviewed code
+  that exists in the code repository.
+
+- [Bitcoin Core #16899][] adds a `dumptxoutset` RPC that will write a
+  copy of the current UTXO set to disk in a serialization format
+  designed for future use with nodes bootstrapped using assumeutxo (see
+  [Newsletter #41][news41 assumeutxo]).  Additionally, a script is added to the
```

Comment:
Needs to link to `#discussion-about-an-assumed-valid-mechanism-for-utxo-snapshots`